File: 1582-special-positions-in-a-binary-matrix/1582-special-positions-in-a-binary-matrix.java
class Solution {
    public int numSpecial(int[][] mat) {
        int[][] cache = new int[Math.max(mat.length, mat[0].length)][2];
        for(int i =0; i< mat.length; ++i) {
            for(int j =0; j< mat[0].length; ++j) {
                if(mat[i][j] == 1) {
                    cache[i][0]++;
                    cache[j][1]++;
                }
            }
        }
        int res = 0;
        for(int i =0; i < mat.length; ++i) {
            for(int j =0; j< mat[0].length; ++j) {
                if(mat[i][j] == 1 && cache[i][0] == 1 && cache[j][1] == 1) {
                    res++;
                }
            }
        }
        return res;
    }
}

File: 1582-special-positions-in-a-binary-matrix/1582-special-positions-in-a-binary-matrix.cpp
class Solution {
public:
    int numSpecial(vector<vector<int>>& mat) {
        int m =mat.size();
        int n = mat[0].size();
        vector<int>Row(m,0);
        vector<int>Col(n,0);
        for(int row =0;row<m;row++)
        {
            for(int col =0;col<n;col++)
            {
                if(mat[row][col]==1)
                {
                    Row[row]++;
                    Col[col]++;

                }
            }
        }
        int ans=0;
        for(int row =0;row<m;row++)
        {
            for(int col =0;col<n;col++)
            {
                if(mat[row][col]==1 && Row[row]==1 && Col[col]==1)
                {
                     
                    ans++;
                }
            }
        }
        return ans;
    }
};